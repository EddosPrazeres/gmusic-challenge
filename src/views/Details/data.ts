export default interface IDetails {
    data?: any;
}
  
export interface IDetailsLayout extends IDetails {
    title: any;
  }
  