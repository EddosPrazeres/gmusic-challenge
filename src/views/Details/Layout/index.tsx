import * as React from "react";
import { IDetailsLayout, JSONUsers } from "../data";
import { CardUserName, Header, CardDetails, CardWrapper, Text, CardElement, ListWrapper, RepositoriesList } from '../../../components'
import './styles.scss'

export default function DetailsView({ title= "DetailsView" }: IDetailsLayout) {
  return (
    <div className="DetailsView">
      <Header title="Detalhes de usuÃ¡rio do Github" hasBack/> 
      <CardUserName {...JSONUsers[0]} />
      <CardDetails />
      <CardDetails title="Repositorios"/>
      <RepositoriesList ListElements={[0,1,2,3,4,5,6,7,8,9]} />
    </div>
  );
}
