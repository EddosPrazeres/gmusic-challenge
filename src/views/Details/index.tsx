import React from "react";
import IDetails from "./data";
import Layout from "./Layout";

export default function DetailsView(props: IDetails) {
  return <Layout title="DetailsView"/>;
}
