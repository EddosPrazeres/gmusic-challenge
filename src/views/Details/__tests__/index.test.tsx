import React from 'react'
import { mount, shallow } from 'enzyme'
import Details from '../index'

const createComponent = (props, useDefault = false) => {
  const componentProps = {  
    ...props,
  }

  const wrapper = useDefault ? mount(<Details  {...componentProps} />) : shallow(<Details     {...componentProps} />)
  const unmount = () => wrapper.unmount()

  return [wrapper, unmount]
}

describe('Details View', () => {
  it.skip('should match snapshot', () => {
    const [wrapper, unmount] = createComponent({})

    expect(wrapper).toMatchSnapshot()
    unmount()
  })
})
