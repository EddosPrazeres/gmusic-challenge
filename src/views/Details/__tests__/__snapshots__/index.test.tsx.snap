// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details View should match snapshot 1`] = `ShallowWrapper {}`;
